presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -90,22 +95,27 @@
 
     private final AtomicReference<SettableFuture<?>> driverBlockedFuture = new AtomicReference<>();
 
+    private final AtomicReference<Optional<Iterator<Page>>> cachedResult = new AtomicReference<>(Optional.empty());
+    private final AtomicReference<Split> split = new AtomicReference<>();
+    private final List<Page> outputPages = new ArrayList<>();
+
     private enum State
     {
         ALIVE, NEED_DESTRUCTION, DESTROYED
     }
 
-    public static Driver createDriver(DriverContext driverContext, List<Operator> operators)
+    public static Driver createDriver(DriverContext driverContext, FragmentResultCacheManager fragmentResultCacheManager, boolean fragmentResultCachingEnabled, List<Operator> operators)
     {
         requireNonNull(driverContext, "driverContext is null");
+        requireNonNull(fragmentResultCacheManager, "fragmentResultCacheManager is null");
         requireNonNull(operators, "operators is null");
-        Driver driver = new Driver(driverContext, operators);
+        Driver driver = new Driver(driverContext, fragmentResultCacheManager, fragmentResultCachingEnabled, operators);
         driver.initialize();
         return driver;
     }
 
     @VisibleForTesting
-    public static Driver createDriver(DriverContext driverContext, Operator firstOperator, Operator... otherOperators)
+    public static Driver createDriver(DriverContext driverContext, FragmentResultCacheManager fragmentResultCacheManager, boolean fragmentResultCachingEnabled, Operator firstOperator, Operator... otherOperators)
```

Comment:
put FragmentResultCacheManager & fragmentResultCachingEnabled in dirverContext?

Reply:
I thought about it and then felt this would impact how the `Driver` works so it is more than a context. But since we already have plan in the context, maybe it does not hurt to put these into context as well. I'll take a look.